### PR TITLE
[Enhancement]: Deduce and maintain expression nullable properties. #4290

### PR DESCRIPTION
**Pull request by badboynt1**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #4290

## What this PR does / why we need it:
Deduce and maintain expression nullable properties.

### CODE REVIEW

**Review by aunjgr — APPROVED** [review]

**Review by reusee — APPROVED** [review]

File: pkg/sql/plan/function/operators.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func initOperators() {
 var operators = map[int]Functions{
 	ISTRUE: {
 		Id:     ISTRUE,
-		Flag:   plan.Function_STRICT,
+		Flag:   plan.Function_PRODUCE_NO_NULL,
```
Comment (fengttt) [inline pkg/sql/plan/function/operators.go:43]: is true does not produce null?   istrue(null) is false?   I am not sure.

**Review by fengttt — COMMENTED** [review]
Do we have real use of this in optimizer or execution?

File: pkg/sql/plan/function/operators.go (hunk 1 above)
Comment (badboynt1) [reply pkg/sql/plan/function/operators.go:43]: i have tested in mo and mysql, it's always false

MySQL [mo]> select null is true;
+--------------+
| null is true |
+--------------+
| false        |
+--------------+
1 row in set (0.00 sec)

MySQL [mo]> select null is false;
+---------------+
| null is false |
+---------------+
| false         |
+---------------+
1 row in set (0.00 sec)

**Comment by badboynt1** [comment]
> Do we have real use of this in optimizer or execution?

issue#3312 depends on this to implement mark join

**Review by nnsgmsone — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

**Review by fengttt — APPROVED** [review]

**Pull request merged**